lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -1009,6 +1009,8 @@ def install_puppetlabs_release_repo_on( hosts, repo = nil, opts = options )
             end
             configure_type_defaults_on( host )
```

Comment:
Would it be better to put the `write_hosts` call in `configure_type_defaults_on` ? I can see arguments for and against this. Until that is settled, I have chosen not to update the unit tests with `write_hosts` in this PR (hence the default stubbing).

Reply:
Though it would work if write_hosts were modified to only update the given hosts in the `HOSTS` entry. I'll probably do that once my ad-hoc run passes.

Reply:
Updated the PR to do this instead.